x/multi-staking/keeper/proposal.go

```diff
@@ -37,6 +40,41 @@ func (k Keeper) AddMultiStakingCoinProposal(
 	return nil
 }
 
+// AddMultiStakingEVMCoinProposal handles the proposals to add a new bond token
+func (k Keeper) AddMultiStakingEVMCoinProposal(
+	ctx sdk.Context,
+	p *types.AddMultiStakingEVMCoinProposal,
+) error {
```

Comment:
If I understand correctly, it looks like we're using the `AddMultiStakingEVMCoinProposal` to both:

Register the contract in the erc20 module, and

Register it in the multistaking store.

This tightly couples the two operations, which means another module wouldn’t be able to register a contract in the erc20 module first and add it to the multistaking module later. I could see us running into a scenario where we want to register the ERC20 contract independently before introducing staking logic.

Would it make sense to decouple the erc20 registration from the multistaking logic to allow for more flexibility?

Reply:
I did that because we don't want to expose erc20 to users like disscussion with @facs95 before. We dont want expose any public funcs of erc20 to user

Also if we do that we will need 2 proposals

Reply:
@hieuvubk thanks - im not sure i fully understand can you ping us about this.